### DISCUSSION TITLE
Show current HEAD (short) hash in fish_git_prompt?

### DISCUSSION THREAD

**Discussion by pjv** [fish-shell/fish-shell#9408] [Q&A] [answered]
This is my current bash prompt when I’m in a git repo:

`pjv-imac:.pjv_bash | master [2354f9f]`

I like being able to see the current abbreviated hash in my prompt.

Is there a simple (configurational) way to replicate this in fish_git_prompt without hacking a fully custom version of that function (that I’d have to maintain across fish updates)?

**Comment by faho** [comment, marked as answer]
The [documentation for fish_git_prompt](https://fishshell.com/docs/current/cmds/fish_git_prompt.html#cmd-fish-git-prompt) lists all of the customization options.

The pertinent one here is $__fish_git_prompt_describe_style:

> $__fish_git_prompt_describe_style can be set to one of the following styles to describe the current HEAD:

And none of the avaibable styles include the hash in addition to the branch, no.

However: This doesn't actually need to be in `fish_git_prompt`! It doesn't touch any of the other bits!

You can just add it on - where you use `fish_git_prompt`, just add it:

```fish
echo -n (prompt_pwd) (fish_git_prompt '%s') '['(git rev-parse HEAD 2>/dev/null | string replace -r '^(.{0,8}).*' '$1')']'
```

`git rev-parse` is cheap, the `string replace` here is a cheesy way to shorten the hash (since `git rev-parse --abbrev-ref` can be slow when you have a lot of commits). The `%s` argument to `fish_git_prompt` removes the braces.

**Reply by pjv** [reply]
Yup, thanks so much for the reply. I had thought of doing something like that in the `fish_prompt.fish`. 

I didn’t show this (should have) but in my current bash prompt, that hash comes after the branch and before stuff like the status, upstream and downstream files, stash info, etc. And so I was looking for a way to replicate how I currently have it in bash, but reflecting on it, I’m totally fine with having the hash be after all the other git-related info - maybe it’s even better that way.

First foray into fish - having fun so far.
